gdpr/service.py

```diff
@@ -0,0 +1,70 @@
+import logging
+from typing import Optional, TYPE_CHECKING
+
+from helsinki_gdpr.types import ErrorResponse
+
+from subscriptions.models import FreeSpotNotificationSubscription
+
+if TYPE_CHECKING:
+    from users.models import User as UserType
+
+logger = logging.getLogger(__name__)
+
+
+def get_user(user: "UserType") -> "UserType":
+    """Function used by the Helsinki Profile GDPR API to get the "user"
+    instance from the "GDPR Model"
+    instance. Since in our case the GDPR Model
+    and the user are one and the same, we simply return
+    the same User instance that is given as a parameter.
+
+    Args:
+        user (UserType): the User instance whose GDPR dat ais being queried
+
+    Returns:
+        UserType: the same User instance
+
+    References:
+        https://github.com/City-of-Helsinki/helsinki-profile-gdpr-api/blob/808dcd30a745f6d18cdf36ccaf07b0cd25844ab0/README.md.
+        https://github.com/City-of-Helsinki/kerrokantasi/blob/256134d4049f0bd1598d59caaebbb813be2d7d9c/kerrokantasi/gdpr.py.
+    """
+    logger.info(f"GDPR data request called for user '{user.uuid}'.")
+    return user
+
+
+def clear_data(user: "UserType", dry_run: bool) -> Optional[ErrorResponse]:
```

Comment:
Is the `dry_run` just for informational purposes here? I mean you're now doing anything with it in the function. Does is roll back a transaction in the helsinki-gdpr-api code to make a dry run?

Reply:
Yea, it's an argument that is available in the interface. It is true that we are not currently doing anything with it, but if it is given to the function that calls the clear_data, [it will raise an exception and the atomic changes will be rolled back](https://github.com/City-of-Helsinki/helsinki-profile-gdpr-api/blob/main/helsinki_gdpr/views.py#L130-L149). I think it is good to have the unused argument there just to give a hint that there is one that the caller can use.